### ISSUE TITLE
Request: add `bref package` command (c.f. `serverless package`)

### ISSUE THREAD

**Issue by oppiansteve** [mnapoli/bref#97]
Would you be averse to me investigating/creating a pull request for this?

This is how I use `serverless package` for CI/CD deployments...

I use `serverless package` so that I can build the package(s) as artefacts in my CI system (currently Bitbucket pipelines).
All commit builds get automatically deployed to dev.
And all tagged commit builds get deployed to stage.
And all tagged commit builds can be manually deployed to prod.

To achieve this, I package all three options at build time.
This is not ideal, but does mean we are deploying to prod the same npm/composer versions etc. that we are testing on dev/stage at least.

Then I just do `serverless deploy --package` to deploy the relevant package.

**Comment by mnapoli** [comment]
Hi, I'm currently exploring a new foundation for Bref (#99), so I'd rather keep the external API surface ("what can be done with bref") limited for now.

To better understand your need: why do you want to package separately from deployment?

**Comment by oppiansteve** [comment]
So that we can test on dev/stage exactly what is to be deployed to prod later (after tests are passed and signed-off).

Rebuilding, even using the NPM lock file doesn't guarantee that it will pull in the exact same code as you tested.

**Comment by oppiansteve** [comment]
plus have an explicit record of what was deployed for debugging/security audits

**Comment by mnapoli** [comment]
> Rebuilding, even using the NPM lock file doesn't guarantee that it will pull in the exact same code as you tested.

Ohh, is that behavior specific to NPM? With Composer (and the lock file) you have a guarantee to have the same dependencies.

In any case that makes sense. I'll keep this in the TODO list, and with AWS SAM (again #99) this should be simple to do as SAM supports this out of the box.

**Comment by oppiansteve** [comment]
Yep, I should have specified the requirement rather than the potential solution - that's a bug-bear of mine, and I still did it!

I could be wrong, but, for NPM, tags can be moved, or the commits changed and there's no checking at install time that (source) hashes actually match what was needed.  (Again, this might have improved since I last investigated).

Composer is much more resilient to change in this regard.

**Comment by oppiansteve** [comment]
AWS SAM looks very useful and a good choice it seems.  Been meaning to investigate it more.
